src/Umbraco.Web/PropertyEditors/ValueConverters/MultiUrlPickerPropertyConverter.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Logging;
+using Umbraco.Core.Models.PublishedContent;
+using Umbraco.Core.PropertyEditors;
+using Umbraco.Core.PropertyEditors.ValueConverters;
+using Umbraco.Core.Services;
+using Umbraco.Web.Models;
+
+namespace Umbraco.Web.PropertyEditors.ValueConverters
+{
+    [DefaultPropertyValueConverter(typeof(JsonValueConverter))]
+    public class MultiUrlPickerPropertyConverter : PropertyValueConverterBase, IPropertyValueConverterMeta
+    {
+        private readonly IDataTypeService _dataTypeService;
+
+        public MultiUrlPickerPropertyConverter(IDataTypeService dataTypeService)
+        {
+            if (dataTypeService == null) throw new ArgumentNullException("dataTypeService");
+            _dataTypeService = dataTypeService;
+        }
+
+        //TODO: Remove this ctor in v8 since the other one will use IoC
+        public MultiUrlPickerPropertyConverter() : this(ApplicationContext.Current.Services.DataTypeService)
+        {
+        }
+
+        public override bool IsConverter(PublishedPropertyType propertyType)
+        {
+            return propertyType.PropertyEditorAlias.Equals(Constants.PropertyEditors.MultiUrlPickerAlias);
+        }
+
+        public override object ConvertDataToSource(PublishedPropertyType propertyType, object source, bool preview)
+        {
+            if (source == null)
+                return null;
+
+            if (source.ToString().Trim().StartsWith("[") == false)
+                return null;
+
+            try
+            {
+                return JArray.Parse(source.ToString());
+            }
+            catch (Exception ex)
+            {
+                LogHelper.Error<MultiUrlPickerPropertyConverter>("Error parsing JSON", ex);
+            }
+            return null;
+        }
+
+        public override object ConvertSourceToObject(PublishedPropertyType propertyType, object source, bool preview)
+        {
+            var isMultiple = IsMultipleDataType(propertyType.DataTypeId, out var maxNumber);
+            if (source == null)
+                return isMultiple
+                    ? Enumerable.Empty<Link>()
+                    : null;
+
+            //TODO: Inject an UmbracoHelper and create a GetUmbracoHelper method based on either injected or singleton
+            if (UmbracoContext.Current == null)
+                return source;
+
+            var umbHelper = new UmbracoHelper(UmbracoContext.Current);
```

Comment:
`UmbracoHelper` should be lazy initialized: if there are no links (empty array) or all links are external, it's not needed (but initializing is quite a performance issue, especially because it's done for every property on a node).

Reply:
As far as I can tell from the code, newing up the UmbracoHelper is a very cheap operation. Where do the performance issue you talk about stem from?

Reply:
Even so, it's not required and it's another object that has to allocate memory and than be garbage collected.

Reply:
It's probably even better to just use the `UmbracoContext.Current.ContentCache.GetById()` and `UmbracoContext.Current.MediaCache.GetById()`, as `UmbracoHelper` just wraps to these calls!

Reply:
Indeed, no real need to use the helper wrapper 👍